gradio/event_queue.py

```diff
@@ -0,0 +1,276 @@
+from __future__ import annotations
+
+import asyncio
+import json
+import time
+from typing import List, Optional
+
+import fastapi
+from fastapi import WebSocketDisconnect
+from pydantic import BaseModel
+
+from gradio.utils import Request, run_coro_in_background
+
+
+class Estimation(BaseModel):
+    msg: Optional[str] = "estimation"
+    rank: Optional[int] = -1  # waiting duration for the xth rank:
```

Comment:
as we discussed, let's make the variable names extra clear and remove the comments

Reply:
yeah I've changed them in #1909